bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoAbstractHandler.java

```diff
@@ -58,32 +58,34 @@
  *
  * @author Marcel Verpaalen - Initial contribution
  */
+@NonNullByDefault
 public abstract class MiIoAbstractHandler extends BaseThingHandler implements MiIoMessageListener {
     protected static final int MAX_QUEUE = 5;
 
-    protected ScheduledFuture<?> pollingJob;
+    protected @Nullable ScheduledFuture<?> pollingJob;
     protected MiIoBindingConfiguration configuration;
     protected MiIoDevices miDevice = MiIoDevices.UNKNOWN;
     protected boolean isIdentified;
 
     protected JsonParser parser;
-    protected byte[] token;
+    protected byte[] token = new byte[0];
 
-    protected MiIoAsyncCommunication miioCom;
+    protected @Nullable MiIoAsyncCommunication miioCom;
     protected int lastId;
 
     protected Map<Integer, String> cmds = new ConcurrentHashMap<Integer, String>();
-    protected ExpiringCache<String> network;
+    protected @Nullable ExpiringCache<String> network;
     protected static final long CACHE_EXPIRY = TimeUnit.SECONDS.toMillis(5);
     protected static final long CACHE_EXPIRY_NETWORK = TimeUnit.SECONDS.toMillis(60);
 
     private final Logger logger = LoggerFactory.getLogger(MiIoAbstractHandler.class);
     protected MiIoDatabaseWatchService miIoDatabaseWatchService;
 
-    @NonNullByDefault
-    public MiIoAbstractHandler(Thing thing) {
+    public MiIoAbstractHandler(Thing thing, MiIoDatabaseWatchService miIoDatabaseWatchService) {
         super(thing);
         parser = new JsonParser();
+        this.miIoDatabaseWatchService = miIoDatabaseWatchService;
+        this.configuration = getConfigAs(MiIoBindingConfiguration.class);
```

Comment:
I don't think you should grab the configuration until the handler is initialized, especially since configurations can change.

Reply:
ok, makes sense. removed it